airflow/models.py

```diff
@@ -4734,9 +4734,21 @@ def __repr__(self):
     def get_state(self):
         return self._state
 
+    def _fail_unfinished_tasks(self, target_state):
+        # if the url has `dagrun/edit` in it then this was
+        # a manual state change kicked off through the UI.
+        # In that case we want to kill any tasks that are unfinished
+        # by setting their current state to the state given by the user
+        if 'dagrun/edit/' in request.url:
```

Comment:
Much cleaner!

![ooo, yeah! Dig it!](https://memegenerator.net/img/instances/55025844.jpg)